base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/table/ActionsTable.java

```diff
@@ -97,11 +97,22 @@ private ScriptGeneratorAction createAction(Map<JavaActionParameter, String> para
 	}
 	
 	private ScriptGeneratorAction createDefaultAction() {
+		return createDefaultAction(null);
+	}
+	
+	private ScriptGeneratorAction createDefaultAction(Integer insertionLocation) {
 		var parametersMap = new HashMap<JavaActionParameter, String>();
 		// Make a parameter/string pair for each parameter in the action
 		for (JavaActionParameter actionParameter: this.actionParameters) {
 			if (actionParameter.getCopyPreviousRow() && this.actions.size() > 0) {
-				ScriptGeneratorAction lastRow = this.actions.get(this.actions.size() - 1);
+				int positionOfRowToCopy;
```

Comment:
This method is getting too long (we should be aiming for only 3-4 lines and all at the same level of abstraction) and it is getting more deeply nested. Can we pull the `if(actionParameter...` and code it contains into a helper method or methods.

Reply:
I have refactored it out twice, as i feel even when pulling out the copyPreviousRow logic it's still probably doing too much. If you think it's overkill we can revert the last commit (8833bdb)

Reply:
I think it's great now